### PR TITLE
Add `MaxPool` derivative tests

### PR DESCRIPTION
**Pull request by jon-tow**
#402

### CODE REVIEW

**Comment by eaplatanios** [comment]
Could you please add comments with python code that generates the reference values? You can take a look at the batch normalization tests for an example.

**Comment by jon-tow** [comment]
@eaplatanios There's no python code. It's from paper.

**Comment by eaplatanios** [comment]
I guess that's true about max pool so we could leave it without.However, using a python reference script can generally help us avoid mistakes that can result even from typos. I'm fine either way.

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -230,6 +254,18 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output, expected)
     }
 
+    func testMaxPool3DGradient(){
+        let layer = MaxPool3D<Float>(poolSize: (2, 2, 2), strides: (1, 1, 1), padding: .valid)
+        let input = Tensor(shape: [1, 2, 2, 2, 1], scalars: (0..<8).map(Float.init))
+        let computedGradient = gradient(at: input, layer) { $1($0).sum() }
+        let expectedGradient = Tensor<Float>([[
```
Comment (Shashi456) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        let expectedGradient =  Tensor<Float>(shape: [1, 2, 2, 2, 1),
                                              scalars: [0, 0, 0, 0, 0, 0, 0, 1])
 ```

Comment (Shashi456) [reply Tests/TensorFlowTests/LayerTests.swift]: Just a suggestion for better readability.

Comment (jon-tow) [reply Tests/TensorFlowTests/LayerTests.swift]: I was going to do that but it seemed easier to visualize which indices carry gradient information from above.

**Comment by jon-tow** [comment]
@eaplatanios I'll add the scripts for consistency.

**Comment by jon-tow** [comment]
@rxwei Sorry about that. I was just about to merge branches.
